src/payment/strategies/stripe-upe/stripe-upe-payment-strategy.ts

```diff
@@ -0,0 +1,192 @@
+import { isHostedInstrumentLike } from '../..';
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType, NotInitializedError, NotInitializedErrorType, RequestError } from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { StoreCreditActionCreator } from '../../../store-credit';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import { StripeElement, StripeElements, StripeElementsOptions, StripeUPEClient } from './stripe-upe';
+import StripeUPEPaymentInitializeOptions from './stripe-upe-initialize-options';
+import StripeUPEScriptLoader from './stripe-upe-script-loader';
+
+export default class StripeUPEPaymentStrategy implements PaymentStrategy {
+    private _initializeOptions?: StripeUPEPaymentInitializeOptions;
+    private _stripeUPEClient?: StripeUPEClient;
+    private _stripeElements?: StripeElements;
+    private _stripeElement?: StripeElement;
```

Comment:
I don't think it is necessary to keep a copy of `_stripeElement` as it can be accessed from `_stripeElements`:


```ts
deinitialize(): Promise<InternalCheckoutSelectors> {
  const element = this._stripeElements?.getElement('payment')?.unmount;

  return Promise.resolve(this._store.getState());
}
```